s3_upload/utils/log.py

```
@@ -35,29 +39,81 @@
+        len(old_backup_files),
+        old_backup_files,
+    )
 
-    return logger
+    if old_backup_files:
+        for old_file in old_backup_files:
+            try:
+                os.remove(old_file)
+            except OSError as error:
+                logger.exception(
+                    "Failed to delete old log file %s: %s", old_file, error
```

Ruff (0.8.0)

Redundant exception object included in `logging.exception` call

(TRY401)